Rg diettracker2

rgoolishian opened the pull request:
Created DatabaseConnector and DbBuilder classes and test classes

sparky005 commented:
This looks good to me to, but you're right, I think the pom.xml changes are causing a lot of conflicts right now. This commit and my earlier commit are both showing conflicts, I have a feeling it'll be the pom file again.

rgoolishian committed:
Corrected exception handling issues in DbParser and DbParserTest

rgoolishian committed:
Merge remote-tracking branch 'origin/rgDiettracker2' into rgDiettracker2

Conflicts:
	pom.xml

rgoolishian committed:
cleaning up conflicts

smarks commented on src/main/java/edu/uml/diet/DatabaseConnector.java:
```diff
@@ -0,0 +1,87 @@
+        this.JDBC_DRIVER = JDBC_DRIVER;
+    }
+
+    public String getHOST_URL() {
```
camel methods names getHostUrl() 

http://www.oracle.com/technetwork/java/codeconventions-150003.pdf

smarks commented on src/main/java/edu/uml/diet/DbBuilder.java:
```diff
@@ -0,0 +1,150 @@
+            resultSet.close();
+        }
+        finally{
+            if(stmt!=null)
```
whole word variable names

smarks commented on src/main/java/edu/uml/diet/DbUserServices.java:
```diff
@@ -0,0 +1,106 @@
+    private String USER = "root";
+    private String PASS = "PASSWORD";
+    public DatabaseConnector databaseConnector = new DatabaseConnector(JDBC_DRIVER,HOST_URL,DB_URL,USER,PASS);
+    public DbBuilder dbBuilder = new DbBuilder(databaseConnector,"diettracker");
```
properties

smarks commented on src/main/java/edu/uml/diet/DbUserServices.java:
```diff
@@ -0,0 +1,106 @@
+     * Method to add new user to database
+     *
+     * @param username
+     * @param password
```
javadoc return

rgoolishian committed:
- Added User class for model of User table in database
- Created DatabaseConnectrException, DuplicateUserException, and PersistenceUserServicesExceptions
- Created Resources directory to house database connection properties xml files
- Overhauled DatabaseBuilder and DatabaseConnector classes to throw new exceptions and use stored database connection properties

rgoolishian committed:
- Completed exception documentation
- Updated pom.xml file

rgoolishian commented:
Completed target changes for this weekend.

sparky005 commented:
Can we merge this? I'm trying to get the login stuff working but the database stuff in master is old
